### PR TITLE
Cursed slot machine now causes IPCs brain damage, still causes humans cloneless

### PR DESCRIPTION
**Pull request by St0rmC4st3r**
## About The Pull Request
Makes a nasty surprise for those, who dare to gamble with forces they don't understand.

## Why It's Good For The Game
Makes gambling for a magical D20 more dangerous if unprepared, since the primary accumulated cloneloss damage is easily fixed with a legion core.

## Changelog
:cl:
add: Adds a bit of brain damage when using the cursed slot machine in greed lavaland ruin.
/:cl:

### CODE REVIEW

**Comment by MarkSuckerberg** [comment]
Have you tested it?

**Comment by ike709** [comment]
I mean this *will* work. It's just a matter of balance concerns. I think this is okay though.

**Review by ike709 — APPROVED** [review]

**Comment by St0rmC4st3r** [comment]
It still doesn't fix the IPC having cloneloss issue. They still take cloneloss.
Can someone put "don't merge"? I am sleepy AF right now and can't really do much.

**Comment by zeskorion** [comment]
nah.

the d20 is already a huge risk in and of itself. vendor has a 5% chance to win. 

i think that legion cores can also heal brain damage anyways

**Comment by St0rmC4st3r** [comment]
Okay. Then to prevent IPCs from getting the prize too easily/prevent the IPC cloneloss issue, I am making it inflict some brain damage to IPCs and prevent it from using it, but I am either too sleepy or too dumb to figure out how to make a race check.

**Comment by MarkSuckerberg** [comment]
Working on setting it up for ya

File: code/modules/ruins/objects_and_mobs/sin_ruins.dm [hunk 1]
```diff
@@ -16,6 +16,7 @@
 		return
 	obj_flags |= IN_USE
 	user.adjustCloneLoss(20)
+	user.adjustBrainLoss(5)
```
Comment (MarkSuckerberg) [inline code/modules/ruins/objects_and_mobs/sin_ruins.dm]: Here's code that works:
```
if(isIPC(user))
		user.adjustBrainLoss(20)
	else
		user.adjustCloneLoss(20)
```

Funny thing is, when I tested it, I won on the first try. Go figure.

I then proceeded to roll a nat 20 and became a wizard. I don't even...

Comment (MarkSuckerberg) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: I ROLLED THE DIE DIRECTLY AFTERWARDS AND ROLLED A 20 CMON WHY DOES THIS HAPPEN NOW?!

Comment (St0rmC4st3r) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: Mate, if only I knew what to write as the if() parameter in the first place.... I have zero knowledge on how race checks should be done properly.

Comment (MarkSuckerberg) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: I had just seen the param in another PR, and thought it might work. Just so happens it does.

Comment (MarkSuckerberg) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: er, wait a moment, I never sent the if() param, I'm an idiot

Comment (MarkSuckerberg) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: Uh, whoops, I messed up the markup in my comment. If param is in there now, sorry

Comment (St0rmC4st3r) [reply code/modules/ruins/objects_and_mobs/sin_ruins.dm]: Thanks, mate

**Comment by St0rmC4st3r** [comment]
So, this one removes cloneloss being dealt to IPCs, but makes them have brainloss. This will give IPCs roughly 20 attempts in total before dying permanently, if I am correct on how brain damage works. It is even nastier because brain damage has no impact on the health meter.

**Review by MarkSuckerberg — APPROVED** [review]
Works as I'd like it to.

File: code/modules/ruins/objects_and_mobs/sin_ruins.dm [hunk 2]
```diff
@@ -14,8 +14,13 @@
 		return
 	if(obj_flags & IN_USE)
 		return
+	if(isIPC(user))
+		user.visible_message("<span class='warning'> As[user] tries to pull [src]'s lever, the machine seems to hesitate a bit.</span>", "<span class='warning'>You feel as if you are trying to put at stake something you don't even have...\ You suddenly feel your mind... Suboptimal?</span>")
```
Comment (ike709) [inline code/modules/ruins/objects_and_mobs/sin_ruins.dm]:
```suggestion
		user.visible_message("<span class='warning'> As [user] tries to pull \the [src]'s lever, the machine seems to hesitate a bit.</span>", "<span class='warning'>You feel as if you are trying to put at stake something you don't even have...\ You suddenly feel your mind... Suboptimal?</span>")
```

**Pull request merged**